### PR TITLE
add grpc gateway for update config

### PR DESCRIPTION
**Pull request by rleungx**
This PR uses the gRPC gateway for updating config.

### CODE REVIEW

File: proto/configpb.proto [hunk 1]
```diff
@@ -12,7 +13,9 @@ option (rustproto.lite_runtime_all) = true;
 service Config {
     rpc Create(CreateRequest) returns (CreateResponse) {}
     rpc Get(GetRequest) returns (GetResponse) {}
-    rpc Update(UpdateRequest) returns (UpdateResponse) {}
+    rpc Update(UpdateRequest) returns (UpdateResponse) {
+        option (google.api.http) = { post: "/pd/api/v1/config" body: "*" };
```
Comment (overvenus) [inline proto/configpb.proto]:
```suggestion
        option (google.api.http) = { post: "/config" body: "*" };
```

I think just `/config` is enough, it's better to not leak PD and api version to protobuf definition

File: proto/configpb.proto [hunk 2]
```diff
@@ -12,7 +13,9 @@ option (rustproto.lite_runtime_all) = true;
 service Config {
     rpc Create(CreateRequest) returns (CreateResponse) {}
     rpc Get(GetRequest) returns (GetResponse) {}
```
Comment (overvenus) [inline proto/configpb.proto]: Please add gateway for other methods too.

File: .circleci/config.yml [hunk 3]
```diff
@@ -16,6 +16,9 @@ jobs:
             echo 'export PATH=$HOME/.local/bin:$PATH' >> $BASH_ENV
             curl -OL https://github.com/google/protobuf/releases/download/v$PROTOC_VERSION/protoc-$PROTOC_VERSION-linux-x86_64.zip
             unzip protoc-$PROTOC_VERSION-linux-x86_64.zip -d $HOME/.local
+      - run:
+          name: Install protoc-gen-grpc-gateway
+          command: go get -u github.com/grpc-ecosystem/grpc-gateway/protoc-gen-grpc-gateway
```
Comment (overvenus) [inline .circleci/config.yml]: Could you use a specific version?

File: proto/configpb.proto (hunk 2 above)
Comment (rleungx) [reply proto/configpb.proto]: This PR is only for compatibility.

**Pull request closed without merging**